test/unit/FutarchyChallenge.js

```diff
@@ -0,0 +1,259 @@
+    })
+
+    it('sets the correct challenger', async () => {
+      expect(await challenge.challenger()).to.equal(CHALLENGER)
+    })
+
+    it('sets the correct stake amount', async () => {
+      expect((await challenge.stakeAmount()).toNumber()).to.equal(STAKE_AMOUNT)
+    })
+
+    it('sets the correct trading period', async () => {
+      expect((await challenge.tradingPeriod()).toNumber()).to.equal(TRADING_PERIOD)
+    })
+
+    it('sets the correct upperBound', async () => {
+      expect((await challenge.upperBound()).toNumber()).to.equal(UPPER_BOUND)
+    })
+
+    it('sets the correct lowerBound', async () => {
+      expect((await challenge.lowerBound()).toNumber()).to.equal(LOWER_BOUND)
+    })
+
+    it('sets the correct futarchyOracleFactory', async () => {
+      expect(await challenge.futarchyOracleFactory()).to.equal(futarchyOracleFactoryMock.address)
+    })
+
+    it('sets the correct scalarPriceOracle', async () => {
+      expect(await challenge.scalarPriceOracle()).to.equal(SCALAR_P_ORACLE_ADDR)
+    })
+
+    it('sets the correct lsmrtMarketMaker', async () => {
+      expect(await challenge.lmsrMarketMaker()).to.equal(LMSR_ADDR)
+    })
+
+    it('sets isStarted to false', async () => {
+      expect(await challenge.isStarted()).to.equal(false)
+    })
+
+    it('sets isFunded to false', async () => {
+      expect(await challenge.isFunded()).to.equal(false)
+    })
+  })
+
+  describe('when deployed with incorrect parameters, ', () => {
+    it('throws if tradingPeriod < 1', async () => {
+      let tradingPeriod = 0
+      await expectRevert(deployChallenge({tradingPeriod}))
+    })
+
+    it('throws if upperBound < lowerBound', async () => {
+      let upperBound = 1
+      let lowerBound = 2
+      await expectRevert(deployChallenge({upperBound, lowerBound}))
+    })
+  })
+
+  describe('ended()', async () => {
+    describe('contingent on futarchOracle outcome', async () => {
+      beforeEach(async () => {
+        challenge = await deployChallenge()
+        await challenge.start()
+        futarchyOracleMock = await FutarchyOracleMock.at(await challenge.futarchyOracle())
+      })
+
+      it('returns true if FutarchyOracle outcome is set', async () => {
+        await futarchyOracleMock.mock_setIsSet(true)
+        expect(await challenge.ended()).to.equal(true)
+      })
+
+      it('returns false if FutarchyOracle outcome is not set', async () => {
+        await futarchyOracleMock.mock_setIsSet(false)
+        expect(await challenge.ended()).to.equal(false)
+      })
+    })
+
+    // TODO: Update tests on fund() / start() conditions when that logic is updated
+  })
+
+  describe('passed()', async () => {
+    beforeEach(async () => {
+      challenge = await deployChallenge()
+      await challenge.start()
+      futarchyOracleMock = await FutarchyOracleMock.at(await challenge.futarchyOracle())
+    })
+
+    it('reverts if ended() returns false', async () => {
```

Comment:
same with these .. I'd generally put the "when" condition in the describe .. like `describe('when ended() returns false'`

I see the benefit of having it on one line though in these cases where there's only one test condition under the describe